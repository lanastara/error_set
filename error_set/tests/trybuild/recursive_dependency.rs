use error_set::error_set;

error_set! {
    MediaError = {
        IoError(std::io::Error)
        } || BookParsingError || DownloadError || UploadError;
    BookParsingError = {
        MissingDescriptionArg
    } || BookSectionParsingError;
    BookSectionParsingError = {
        MissingNameArg,
        NoContents,
    } || BookParsingError;
    DownloadError = {
        CouldNotConnect,
        OutOfMemory(std::io::Error),
    };
    UploadError = {
        NoConnection(std::io::Error),
    };
}