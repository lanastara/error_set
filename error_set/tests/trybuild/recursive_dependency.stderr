error: Cycle Detected: BookParsingError->BookSectionParsingError->BookParsingError
 --> tests/trybuild/recursive_dependency.rs:7:5
  |
7 |     BookParsingError = {
  |     ^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/trybuild/recursive_dependency.rs:21:2
   |
21 | }
   |  ^ consider adding a `main` function to `$DIR/tests/trybuild/recursive_dependency.rs`
